parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -1804,14 +1804,27 @@ private static void copy(SeekableInputStream from, PositionOutputStream to, long
    * @throws IOException if there is an error while writing
    */
   public void end(Map<String, String> extraMetaData) throws IOException {
+    final long footerStart = out.getPos();
+
+    // Build the footer metadata) in memory using the helper stream
+    InMemoryPositionOutputStream buffer = new InMemoryPositionOutputStream(footerStart);
```

Comment:
I don't think adding yet another buffered stream help anything here. The output stream implementation should already has its internal buffer and eventually it will flush data to the underlying storage which means network or filesystem error may still happen.

Reply:
I don't think so. I believe many OutputStream implementations (especially FSDataOutputStream) should have already implemented their internal buffers. Perhaps @steveloughran may correct me if I was wrong.

Even if we want to minimize the IO count, this is the wrong place since it only contains the footer.

Reply:
I agree with @wgtmac that this doesn't solve the problem, and also introduces other issues by introducing another buffer. Since writes to object stores are pretty expensive, those libraries often pre-allocate a rather big buffer. Adding another buffer as suggested in the PR will introduce more memory pressure, which I think should be avoided. This problem will amplify if you write to a lot of files in parallel.

Reply:
These solutions are all sound like workaround for avoiding failures but actually those failures may eventually emerge. I think what the downstream projects need is to know the writer has failed and gracefully handle the error (not to get yet another exception).